ilqr/cost.py

```diff
@@ -353,7 +353,7 @@ def __init__(self, f, state_size, action_size, **kwargs):
         # Prepare inputs.
         self._x = x = T.dvector("x")
         self._u = u = T.dvector("u")
-        self._i = i = T.dvector("i")
+        self._i = i = T.dscalar("i")
```

Comment:
I think we should keep `T.dvector` here to be consistent with the rest of the `Cost` and `Dynamics` implementations. The idea was to support computing the cost of a batch of (x, u, i) pairs in one shot. We can revisit this though, but probably better in a separate PR.

Reply:
Should `terminal` also be a vector if `i` is? I'm confused how these interact.